metpy/tests/test_deprecation.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) 2018 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""Test functionality of MetPy's deprecation code."""
+
+import pytest
+
+from metpy import deprecation
+
+
+class FakeyMcFakeface(object):
+    """Our faked object for testing."""
+
+    @classmethod
```

Comment:
I'm not sure this actually works, based on matplotlib/matplotlib#12151.

Reply:
I am unsure what code you are referencing, but indeed, I found that it is very hard to deprecate a classmethod.  I see that matplotlib's current code suggests a hack ordering workaround of the decorators.